lib/oauth.js

```diff
@@ -503,10 +504,17 @@ internals.Client.prototype.signature = function (method, baseUri, params, oauth,
                        internals.encode(baseUri) + '&' +
                        internals.encode(normalizedParam);
 
-    // HMAC-SHA1 (3.4.2)
-
-    const key = tokenSecret ? (this.settings.clientSecret + internals.encode(tokenSecret)) : this.settings.clientSecret;
-    return Crypto.createHmac('sha1', key).update(baseString).digest('base64');
+    let sig;
+    if (oauth.oauth_signature_method === 'RSA-SHA1') { // RSA-SHA1 (3.4.3)
+        const sign = Crypto.createSign('sha1');
+        sign.update(baseString);
+        sig = sign.sign(this.settings.clientSecret).toString('base64');
```

Comment:
1. Instead of doing `.toString` you can pass the `base64` to the 2nd parameter of sign.
2. I would also suggest you chain the calls to match the code below.
3. Format this slightly differently:
- instead of doing the else, just return from the function early. Then you don't need to create the `let sig` variable and I find we usually try to return early rather than doing if/else stmts.